lib/spack/env/cc

```diff
@@ -98,25 +98,25 @@ case "$command" in
     cpp)
         mode=cpp
         ;;
-    cc|c89|c99|gcc|clang|icc|pgcc|xlc)
+    cc|c89|c99|gcc|clang|icc|pgcc|xlc|xlc_r)
```

Comment:
I'm fine with a new compiler. I hate the idea of hacking the version to include an `_r`. If `xlc` and `xlc_r` work differently and there's any chance that a user would want to use both, we should support that, even if they're the same version.